book-1-orientation/chapters/QUEST.md

```diff
@@ -0,0 +1,287 @@
+
+                    // Note how we access a Adventurer object's property
+                    Adventurer.Awesomness += _awesomenessChange;
+                }
+                else
+                {
+                    Console.WriteLine("You have failed the challenge, there will be consequences.");
+                    Adventurer.Awesomness -= _awesomenessChange;
+                }
+
+                // Note how we call a Adventurer object's method
+                Console.WriteLine(Adventurer.GetAdventurerStatus());
+                Console.WriteLine();
+            }
+        }
+    }
+    ```
+
+1. Read over the code. Imagine what the code will do when it runs. Write some notes about what you think will happen.
+1. Run the program. Does it do what you expected?
+
+> **NOTE:** At the end of each phase you should Add, Commit and Push to GitHub.
+
+### Phase 2
+
+Our adventurer is currently named "Jack". Studies show that "Jack" is probably not the application user's name. Update the code to prompt the user for their name and pass that name to the `Adventurer` constructor when creating the new `Adventurer` object.
+
+### Phase 3
+
+This quest is so much fun that everyone is sure to want to do it more than once. Update the code to ask the user if they'd like to repeat the quest after the it has been completed. If the user says "yes", start the quest over. Otherwise, end the program.
+
+### Phase 4
+
+I don't know if you noticed, but our adventurer is naked. What happened to all that talk about a brightly colored robe and shiny hat?
+
+In this phase we'll add the robe.
+
+1. Create new file called `Robe.cs`.
+1. Inside the file define a `Robe` class.
+1. The class should have the following mutable properties.
+    * Colors - a list of strings to hold the colors of the robe
+    * Length - an integer describing the length of the robe in inches
+1. The class should not contain any methods or constructors.
+1. Add a new immutable property to the `Adventurer` class called `ColorfulRobe`. The type of this property should be `Robe`.
+1. Add a new constructor parameter to the `Adventurer` class to accept a `Robe` and to set the `ColorfulRobe` property.
+1. Add a new method to the `Adventurer` class called `GetDescription`. This method should return a string that contains the adventurer's name and a description of their robe.
+1. In `Program.cs` create a new instance of the `Robe` class and set it's properties.
+1. Pass the new `Robe` into the constructor of the `Adventurer`.
+1. Before the adventurer starts their challenge, call the `GetDescription` method and print the results to the console.
+
+### Phase 5
+
+Let's cover that hatless head.
+
+1. Create a new class called `Hat` in it's own file.
+1. Add a mutable integer property called `ShininessLevel` to indicate how shiny the hat is.
+1. Add a computer string property called `ShininessDescription` to return a text description of the hat's shininess according to the following rules.
```

Comment:
Serious comment: 'computed string property'
Not serious: Dragonball Z fans will appreciate you if you say '9001' should be "blinding" on line 259. But then they'll think you like anime.